src/main/java/fi/aalto/cs/apluscourses/model/FileDoesNotExistException.java

```diff
@@ -24,4 +30,15 @@ public Path getPath() {
   public String getName() {
     return name;
   }
+
+  private void writeObject(ObjectOutputStream oos) throws IOException {
```

Comment:
kinda like this @Aalto-LeTech/apluscourses-plugin-team ?

Reply:
This looks like a nice use of `readObject`/`writeObject` but...

- The string representation returned by `Path::toString` is platform specific (`/` vs `\`). Generating a path from that string may not be possible on another platform. Thus, this serialization from one machine to another may not work.
- This adds a considerable amount of must-be-tested code while providing very little value.

So...


Would it just be safer and easier to store the path as a string?

Reply:
I would in general suggest that we don't store paths as a string just to support serialization which we do not use, this is maybe a special case since it's just an exception but still

Reply:
I'll just make `Path` and `HttpResponse` `transient` 😝